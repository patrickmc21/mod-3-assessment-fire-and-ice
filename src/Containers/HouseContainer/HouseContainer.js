import React from 'react';
import { connect } from 'react-redux';
import HouseCard from '../../components/HouseCard/HouseCard';
import wolf from './images/wolf.gif';

export const HouseContainer = (props) => {
  const { houses } = props;
  const houseCards = houses.map(house => {
    return (<HouseCard key={house.name} house={house} />);
  })
  return (
    <div className='Container'>
      {houses.length < 1 && <img src={wolf} id='Wolf' />}
      {houses.length >= 1 && houseCards}
    </div>
  );
};

export const mapStateToProps = (state) => ({
  houses: state.houses
});


export default connect(mapStateToProps, null)(HouseContainer);