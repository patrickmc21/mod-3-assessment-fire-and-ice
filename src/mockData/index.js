/* eslint-disable */

export const mockHouse = [
{  
   url:"https://www.anapioficeandfire.com/api/houses/81",
   name:"House Corbray of Heart's Home",
   region:"",
   coatOfArms:"Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
   words:"",
   titles:[  
   "Lord of Heart's Home",
   "King of the Fingers (historical)"
   ],
   seats:[  
   "Heart's Home"
   ],
   currentLord:"https://www.anapioficeandfire.com/api/characters/681",
   heir:"https://www.anapioficeandfire.com/api/characters/677",
   overlord:"https://www.anapioficeandfire.com/api/houses/7",
   founded:"",
   founder:"https://www.anapioficeandfire.com/api/characters/256",
   diedOut:"",
   ancestralWeapons:[  
   "Lady Forlorn"
   ],
   cadetBranches:[  

   ],
   swornMembers:[  
   "https://www.anapioficeandfire.com/api/characters/255",
   "https://www.anapioficeandfire.com/api/characters/256",
   "https://www.anapioficeandfire.com/api/characters/447",
   "https://www.anapioficeandfire.com/api/characters/654",
   "https://www.anapioficeandfire.com/api/characters/677",
   "https://www.anapioficeandfire.com/api/characters/681",
   "https://www.anapioficeandfire.com/api/characters/871",
   "https://www.anapioficeandfire.com/api/characters/1182"
   ]
}
];

export const swornMember = {  
   "url":"https://www.anapioficeandfire.com/api/characters/255",
   "name":"Corwyn Corbray",
   "gender":"Male",
   "culture":"",
   "born":"",
   "died":"In 134 AC, at Runestone",
   "titles":[  
      "Ser",
      "Lord Regent"
   ],
   "aliases":[  
      ""
   ],
   "father":"",
   "mother":"",
   "spouse":"https://www.anapioficeandfire.com/api/characters/871",
   "allegiances":[  
      "https://www.anapioficeandfire.com/api/houses/81"
   ],
   "books":[  
      "https://www.anapioficeandfire.com/api/books/11"
   ],
   "povBooks":[  

   ],
   "tvSeries":[  
      ""
   ],
   "playedBy":[  
      ""
   ]
};