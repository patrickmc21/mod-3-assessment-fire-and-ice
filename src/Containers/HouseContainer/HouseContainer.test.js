import React from 'react';
import { shallow } from 'enzyme';
import { HouseContainer, mapStateToProps } from './HouseContainer';
import { mockHouse } from '../../mockData';

describe('HouseContainer', () => {

  it('should match the snapshot if no houses present', () => {
    const wrapper = shallow(
      <HouseContainer houses={[]} />
    );
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should match the snapshot if houses present', () => {
    const wrapper = shallow(
      <HouseContainer houses={mockHouse} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should map houses to props', () => {
    const mockState = {houses: mockHouse};
    const expected = mockHouse;
    const mapped = mapStateToProps(mockState);
    expect(mapped.houses).toEqual(expected);
  });
});