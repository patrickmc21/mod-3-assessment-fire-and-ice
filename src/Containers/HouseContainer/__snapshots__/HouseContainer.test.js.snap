// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseContainer should match the snapshot if houses present 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houses={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": Array [
                "https://www.anapioficeandfire.com/api/characters/255",
                "https://www.anapioficeandfire.com/api/characters/256",
                "https://www.anapioficeandfire.com/api/characters/447",
                "https://www.anapioficeandfire.com/api/characters/654",
                "https://www.anapioficeandfire.com/api/characters/677",
                "https://www.anapioficeandfire.com/api/characters/681",
                "https://www.anapioficeandfire.com/api/characters/871",
                "https://www.anapioficeandfire.com/api/characters/1182",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [
          <HouseCard
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Lady Forlorn",
                                    ],
                                    "cadetBranches": Array [],
                                    "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                                    "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
                                    "diedOut": "",
                                    "founded": "",
                                    "founder": "https://www.anapioficeandfire.com/api/characters/256",
                                    "heir": "https://www.anapioficeandfire.com/api/characters/677",
                                    "name": "House Corbray of Heart's Home",
                                    "overlord": "https://www.anapioficeandfire.com/api/houses/7",
                                    "region": "",
                                    "seats": Array [
                                      "Heart's Home",
                                    ],
                                    "swornMembers": Array [
                                      "https://www.anapioficeandfire.com/api/characters/255",
                                      "https://www.anapioficeandfire.com/api/characters/256",
                                      "https://www.anapioficeandfire.com/api/characters/447",
                                      "https://www.anapioficeandfire.com/api/characters/654",
                                      "https://www.anapioficeandfire.com/api/characters/677",
                                      "https://www.anapioficeandfire.com/api/characters/681",
                                      "https://www.anapioficeandfire.com/api/characters/871",
                                      "https://www.anapioficeandfire.com/api/characters/1182",
                                    ],
                                    "titles": Array [
                                      "Lord of Heart's Home",
                                      "King of the Fingers (historical)",
                                    ],
                                    "url": "https://www.anapioficeandfire.com/api/houses/81",
                                    "words": "",
                                  }
            }
/>,
        ],
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": "House Corbray of Heart's Home",
        "nodeType": "function",
        "props": Object {
          "house": Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "cadetBranches": Array [],
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
            "diedOut": "",
            "founded": "",
            "founder": "https://www.anapioficeandfire.com/api/characters/256",
            "heir": "https://www.anapioficeandfire.com/api/characters/677",
            "name": "House Corbray of Heart's Home",
            "overlord": "https://www.anapioficeandfire.com/api/houses/7",
            "region": "",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": Array [
              "https://www.anapioficeandfire.com/api/characters/255",
              "https://www.anapioficeandfire.com/api/characters/256",
              "https://www.anapioficeandfire.com/api/characters/447",
              "https://www.anapioficeandfire.com/api/characters/654",
              "https://www.anapioficeandfire.com/api/characters/677",
              "https://www.anapioficeandfire.com/api/characters/681",
              "https://www.anapioficeandfire.com/api/characters/871",
              "https://www.anapioficeandfire.com/api/characters/1182",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
            "url": "https://www.anapioficeandfire.com/api/houses/81",
            "words": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <HouseCard
              house={
                            Object {
                                          "ancestralWeapons": Array [
                                            "Lady Forlorn",
                                          ],
                                          "cadetBranches": Array [],
                                          "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                                          "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
                                          "diedOut": "",
                                          "founded": "",
                                          "founder": "https://www.anapioficeandfire.com/api/characters/256",
                                          "heir": "https://www.anapioficeandfire.com/api/characters/677",
                                          "name": "House Corbray of Heart's Home",
                                          "overlord": "https://www.anapioficeandfire.com/api/houses/7",
                                          "region": "",
                                          "seats": Array [
                                            "Heart's Home",
                                          ],
                                          "swornMembers": Array [
                                            "https://www.anapioficeandfire.com/api/characters/255",
                                            "https://www.anapioficeandfire.com/api/characters/256",
                                            "https://www.anapioficeandfire.com/api/characters/447",
                                            "https://www.anapioficeandfire.com/api/characters/654",
                                            "https://www.anapioficeandfire.com/api/characters/677",
                                            "https://www.anapioficeandfire.com/api/characters/681",
                                            "https://www.anapioficeandfire.com/api/characters/871",
                                            "https://www.anapioficeandfire.com/api/characters/1182",
                                          ],
                                          "titles": Array [
                                            "Lord of Heart's Home",
                                            "King of the Fingers (historical)",
                                          ],
                                          "url": "https://www.anapioficeandfire.com/api/houses/81",
                                          "words": "",
                                        }
              }
/>,
          ],
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "House Corbray of Heart's Home",
          "nodeType": "function",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": Array [
                "https://www.anapioficeandfire.com/api/characters/255",
                "https://www.anapioficeandfire.com/api/characters/256",
                "https://www.anapioficeandfire.com/api/characters/447",
                "https://www.anapioficeandfire.com/api/characters/654",
                "https://www.anapioficeandfire.com/api/characters/677",
                "https://www.anapioficeandfire.com/api/characters/681",
                "https://www.anapioficeandfire.com/api/characters/871",
                "https://www.anapioficeandfire.com/api/characters/1182",
              ],
              "titles": Array [
                "Lord of Heart's Home",
                "King of the Fingers (historical)",
              ],
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HouseContainer should match the snapshot if no houses present 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houses={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          id="Wolf"
          src="wolf.gif"
/>,
        false,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "Wolf",
          "src": "wolf.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            id="Wolf"
            src="wolf.gif"
/>,
          false,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "Wolf",
            "src": "wolf.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
